### ISSUE TITLE
Add detection of "The truth value of a DataFrame is ambiguous."

### ISSUE THREAD

**Issue by karolzlot** [microsoft/pyright#5024]
**Is your feature request related to a problem? Please describe.**
Currently this code is considered by pyright and pylance as valid (no diagnostics, even in strict mode):

```python
import pandas

df = pandas.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})

if df == "foo":
    print("foo") 

```

But this code doesn't work:
![image](https://user-images.githubusercontent.com/11590960/234967962-749f7a05-fe32-4ffa-b379-21681166f7ad.png)

**Describe the solution you'd like**
It would help if pyright could detect this bug.

**Additional context**
n/a

**Comment by erictraut** [comment]
Pyright is a standards-compliant static type checker. It doesn't have any special knowledge about the internal semantics, behaviors, or error conditions of third-party libraries like pandas. There isn't any static type violation in this code, so pyright is correct in not generating an error here.
